test/unit/dao/FeiDao.test.ts

```diff
@@ -0,0 +1,189 @@
+import { expectRevert, time, getCore, getAddresses } from '../../helpers';
+import { expect } from 'chai';
+import hre, { artifacts, ethers, network } from 'hardhat';
+import { Signer } from 'ethers';
+import { TransactionResponse } from '@ethersproject/providers';
+
+const FeiDAO = artifacts.readArtifactSync('FeiDAO');
+const Timelock = artifacts.readArtifactSync('Timelock');
+const Tribe = artifacts.readArtifactSync('Tribe');
+
+const toBN = ethers.BigNumber.from;
+
+describe('FeiDAO', function () {
+  let userAddress: string;
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [addresses.userAddress];
+
+    for (const address of impersonatedAddresses) {
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [address]
+      });
+
+      impersonatedSigners[address] = await ethers.getSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ userAddress } = await getAddresses());
+
+    await network.provider.request({
+      method: 'hardhat_reset',
+      params: []
+    });
+
+    this.core = await getCore();
```

Comment:
`this` inside of a mocha test should be avoided; instead, declare vars inside the uppermost "describe" block and reference these to avoid weird scoping issues, especially when nesting describe hooks

source: https://stackoverflow.com/questions/27358338/mocha-this-in-before-and-beforeeach-hooks